src/Compilers/Core/Portable/Syntax/LineMapping.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.Serialization;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Represents a line mapping defined by a single line mapping directive (<c>#line</c> in C# or <c>#ExternalSource</c> in VB).
+    /// </summary>
+    [DataContract]
```

Comment:
>DataContract [](http://example.com/codeflow?start=5&length=12)

Just curious: where are we serializing `LineMapping`?

Reply:
We actually don't. Will remove.